From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Sun, 21 Jan 2018 21:04:37 +0100
Subject: networkAgent: Pick up VPN service dir from pkg-config

It turns out that NetworkManager does export the directory as pkg-config
variable after all, so use that instead of building the path ourselves
from the prefix.

https://bugzilla.gnome.org/show_bug.cgi?id=789811
---
 js/misc/config.js.in             |  2 +-
 js/misc/meson.build              |  2 +-
 js/ui/components/networkAgent.js |  2 +-
 meson.build                      | 11 ++++-------
 4 files changed, 7 insertions(+), 10 deletions(-)

diff --git a/js/misc/config.js.in b/js/misc/config.js.in
index f296275..28bff96 100644
--- a/js/misc/config.js.in
+++ b/js/misc/config.js.in
@@ -14,6 +14,6 @@ var GETTEXT_PACKAGE = '@GETTEXT_PACKAGE@';
 var LOCALEDIR = '@datadir@/locale';
 /* other standard directories */
 var LIBEXECDIR = '@libexecdir@';
-var NMPREFIXDIR = '@nmprefixdir@';
+var VPNDIR = '@vpndir@';
 /* g-i package versions */
 var LIBMUTTER_API_VERSION = '@LIBMUTTER_API_VERSION@'
diff --git a/js/misc/meson.build b/js/misc/meson.build
index bbdc041..5a48717 100644
--- a/js/misc/meson.build
+++ b/js/misc/meson.build
@@ -7,7 +7,7 @@ jsconf.set10('HAVE_BLUETOOTH', bt_dep.found())
 jsconf.set10('HAVE_NETWORKMANAGER', have_networkmanager)
 jsconf.set('datadir', datadir)
 jsconf.set('libexecdir', libexecdir)
-jsconf.set('nmprefixdir', nm_prefix)
+jsconf.set('vpndir', vpndir)
 
 config_js = configure_file(
   input: 'config.js.in',
diff --git a/js/ui/components/networkAgent.js b/js/ui/components/networkAgent.js
index 9c3449f..fd9d9bb 100644
--- a/js/ui/components/networkAgent.js
+++ b/js/ui/components/networkAgent.js
@@ -594,7 +594,7 @@ var NetworkAgent = new Lang.Class({
         this._vpnRequests = { };
         this._notifications = { };
 
-        this._pluginDir = Gio.file_new_for_path(GLib.build_filenamev([Config.NMPREFIXDIR, 'lib/NetworkManager/VPN']));
+        this._pluginDir = Gio.file_new_for_path(Config.VPNDIR);
         try {
             let monitor = this._pluginDir.monitor(Gio.FileMonitorFlags.NONE, null);
             monitor.connect('changed', () => { this._vpnCacheBuilt = false; });
diff --git a/meson.build b/meson.build
index f8dc155..32b34c9 100644
--- a/meson.build
+++ b/meson.build
@@ -112,6 +112,8 @@ if enable_networkmanager != 'no'
     version: secret_req, required: want_networkmanager
   )
 
+  vpndir = nm_deps[0].get_pkgconfig_variable('vpnservicedir')
+
   have_networkmanager = true
   foreach dep : nm_deps
     have_networkmanager = have_networkmanager and dep.found()
@@ -121,14 +123,9 @@ if enable_networkmanager != 'no'
     nm_deps = []
   endif
 else
-  have_networkmanager = false
-endif
+  vpndir = prefix
 
-networkmanager_dep = dependency('NetworkManager', required: false)
-if (networkmanager_dep.found())
-  nm_prefix = networkmanager_dep.get_pkgconfig_variable('prefix')
-else
-  nm_prefix = prefix
+  have_networkmanager = false
 endif
 
 enable_systemd = get_option('enable-systemd')
